addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -92,7 +92,20 @@ export class SelfOrder extends Reactive {
             this.onNotified("PAYMENT_STATUS", ({ payment_result, data }) => {
                 if (payment_result === "Success") {
                     this.models.replaceDataByKey("uuid", data);
-                    this.router.navigate("confirmation");
+                    const order = this.models["pos.order"].find(
+                        (o) => o.access_token === data["pos.order"][0].access_token
+                    );
+                    if (["paid", "invoiced", "done"].includes(order?.state)) {
+                        this.notification.add(_t("Your order has been paid"), {
+                            type: "success",
+                        });
+                    }
+                    if (!this.currentOrder.access_token && order && order.access_token) {
+                        this.router.navigate("confirmation", {
+                            orderAccessToken: order.access_token,
+                            screenMode: "order",
+                        });
```

Comment:
I would like to create a generic method in `pos_self_order` and adapt in `pos_online_payment_self_order`
i.e.
```
    finalizeOrder(accessToken, screenMode="order") {
        this.router.navigate("confirmation", {
            orderAccessToken: accessToken,
            screenMode: screenMode,
        });
    },
```

Reply:
> I would like to create a generic method in `pos_self_order` and adapt in `pos_online_payment_self_order` i.e.
> 
> ```
>     finalizeOrder(accessToken, screenMode="order") {
>         this.router.navigate("confirmation", {
>             orderAccessToken: accessToken,
>             screenMode: screenMode,
>         });
>     },
> ```

Thanks for your suggestion 
we can use a generic method async confirmationPage instead of creating new one 

``` 
async confirmationPage(screen_mode, device, access_token = "") {
        this.router.navigate("confirmation", {
            orderAccessToken: access_token || this.currentOrder.access_token,
            screenMode: screen_mode,
        });
        if (device === "kiosk") {
            this.printKioskChanges();
        }
    }
```

what do you think ?

Reply:
Yes, It's LGTM just need to test the print!